src/modules/frequency/application/frequency.service.ts

```diff
@@ -1,26 +1,258 @@
+  FREQUENCY_QUERIES_TOKEN,
+  IFrequencyQueries,
+} from "./frequency.service.query.interfaces";
+import {
+  UserClassesResponseDTO,
+  UpdateGeneralAttendanceRequestDTO,
+  UpdateGeneralAttendanceItemDTO,
+  StudentListByClassAndDateResponseDTO,
+  UpdateClassAttendanceRequestDTO,
+  GeneralAttendanceResponseDTO,
+} from "./dtos";
+import { Frequency } from "../domain/frequency.entity";
+import {
+  FREQUENCY_REPOSITORY_TOKEN,
+  IFrequencyRepository,
+} from "../domain/frequency.repository";
+import { FrequencyStatus } from "src/common/enums/domain.enums";
 
 @Injectable()
-export class FrequencyService{
-    @Inject(FREQUENCY_QUERIES_TOKEN)
-    private readonly frequencyQueryService: IFrequencyQueries;
-    public async getUserClasses(userId: number): Promise<ResponseGetMyClassesDTO>{
-        var userClasses = await this.frequencyQueryService.getMyClasses(userId);
-        userClasses.push({
-            classId: null,
-            className: "Geral",
-            classState: "ATIVA",
-            levelName: null,
-            isGeral: true,
-            activity: {
-                activityId: null,
-                activityName: "Atividade geral",
-            },
-        })
-        return {
-            classes : userClasses
-        };
-    }
-}
\ No newline at end of file
+export class FrequencyService {
+  @Inject(FREQUENCY_QUERIES_TOKEN)
+  private readonly frequencyQueryService: IFrequencyQueries;
+  @Inject(FREQUENCY_REPOSITORY_TOKEN)
+  private readonly frequencyRepository: IFrequencyRepository;
+
+  public async getUserClasses(userId: number): Promise<UserClassesResponseDTO> {
+    const userClasses = await this.frequencyQueryService.getMyClasses(userId);
+    userClasses.push({
+      classId: null,
+      className: "Geral",
+      classState: "ATIVA",
+      levelName: null,
+      isGeral: true,
+      activity: {
+        activityId: null,
+        activityName: "Atividade geral",
+      },
+    });
+    return {
+      classes: userClasses,
+    };
+  }
+  public async getGeneralAttendance(
+    date: Date,
+  ): Promise<GeneralAttendanceResponseDTO> {
+    const studentList =
+      await this.frequencyQueryService.getGeneralAttendance(date);
+    return {
+      date: date.toISOString().split("T")[0],
+      studentList: studentList,
+    };
+  }
+  public async updateGeneralAttendance(
+    object: UpdateGeneralAttendanceRequestDTO,
+  ): Promise<boolean> {
+    const date = object.date;
+    const isConsistent = await this.isGeneralAttendanceConsistent(
+      object["studentList"],
+      date,
+    );
+    if (!isConsistent.isValid) {
+      throw new BadRequestException(
+        `Attendance permission conflict: The generalAttendanceAllowed values in your request don't match our system records. Problem with studentId: ${isConsistent.problematicId}`,
+      );
+    }
+    const validItems: UpdateGeneralAttendanceItemDTO[] = object[
+      "studentList"
+    ].filter((item) => item.generalAttendanceAllowed === true);
```

Comment:
Redundant, but its fine either way
```suggestion
    ].filter((item) => item.generalAttendanceAllowed);
```

Reply:
De fato, fui escrevendo o código, acabei não me ligando.